sdk/java/pulumi/src/main/java/com/pulumi/core/internal/Urn.java

```diff
@@ -148,10 +150,10 @@ public String asString() {
     @ParametersAreNonnullByDefault
     @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
     public static final class QualifiedTypeName {
-        public final Optional<Type> parent;
+        public final Optional<String> parent;
```

Comment:
This is an internal package so I believe it should be safe to change this.

Reply:
Consider clarifying with a comment here that parent is the full "all parents" string, and not just the last parent.
We should consider renaming this to parents if that won't break any callers. (This doesn't look like it's exposed so it should be fine?)